repos/system_upgrade/el7toel8/actors/filterrpmtransactionevents/actor.py

```diff
@@ -36,8 +37,12 @@ def process(self):
 
         to_remove.difference_update(to_keep)
 
+        # run upgrade for the rest of RH signed pkgs which we do not have rule for
+        to_upgrade = installed_pkgs - set(list(to_install) + list(to_remove))
```

Comment:
Sorry if I got something wrong: do you really need a set-list-set conversion here?

Why not
```
pkg_to_process = set(to_install).update(to_remove)
to_upgrade = installed_pkgs - pkg_to_process
```